### PR TITLE
Add --git-cache to bootstrap.py.

### PR DESCRIPTION
**Pull request by spxtr**

### CODE REVIEW

**Comment by rmmh** [comment]
This looks pretty good, though the number of arguments getting passed around is getting a bit ridiculous-- might want to objectify this soon.

Have you done a manual run to ensure that it works properly?

File: jenkins/bootstrap_test.py [hunk 1]
```diff
@@ -994,6 +994,7 @@ def setUp(self):
         ]
         self.root_github = tempfile.mkdtemp()
         self.root_workspace = tempfile.mkdtemp()
+        self.root_git_cache = tempfile.mkdtemp()
```
Comment (fejta) [inline jenkins/bootstrap_test.py]: Do we need this in all test cases given that it is only used in one?

**Comment by spxtr** [comment]
> Have you done a manual run to ensure that it works properly?

Yes, and also ensured that the test is broken if I do things differently.

File: jenkins/bootstrap.py [hunk 2]
```diff
@@ -215,7 +215,15 @@ def checkout(call, repo, branch, pull, ssh=False):
         refs, checkouts = branch_ref(branch)
 
     git = 'git'
-    call([git, 'init', repo])
+    if git_cache:
```
Comment (fejta) [inline jenkins/bootstrap.py]: Personally I would like to see a checkout test for this instead of (or in addition to) the bootstrap test.

Comment (fejta) [inline jenkins/bootstrap.py]: The reason being that smaller tests are better than larger ones, and all the complexity lies in checkout.

File: jenkins/bootstrap_test.py (hunk 1 above)
Comment (spxtr) [inline jenkins/bootstrap_test.py]: No, we only need it for the one.

File: jenkins/bootstrap.py (hunk 2 above)
Comment (spxtr) [inline jenkins/bootstrap.py]: I'll switch it over.

Comment (spxtr) [inline jenkins/bootstrap.py]: Actually I think this belongs under `IntegrationTest`. If I want to switch it to call `checkout` directly then I need to copy the line from `bootstrap` that deals with `_call`. It's not a big deal, but a little scary.

Comment (fejta) [inline jenkins/bootstrap.py]: Ack

**Pull request merged**